pr_agent/algo/git_patch_processing.py

```diff
@@ -108,3 +108,78 @@ def handle_patch_deletions(patch: str, original_file_content_str: str,
                 logging.info(f"Processing file: {file_name}, hunks were deleted")
             patch = patch_new
     return patch
+
+
+def convert_to_hunks_with_lines_numbers(patch: str, file) -> str:
```

Comment:
The `convert_to_hunks_with_lines_numbers` function could be optimized. Currently, it iterates over the patch lines multiple times, which could be inefficient for large patches. Consider refactoring the function to reduce the number of iterations. [medium]